drivers/mipi_dbi/mipi_dbi_stm32_fmc.c

```diff
@@ -0,0 +1,133 @@
+	/* Reset GPIO */
+	const struct gpio_dt_spec reset;
+	/* Power GPIO */
+	const struct gpio_dt_spec power;
+	uint16_t *register_addr;
+	uint16_t *data_addr;
+};
+
+struct mipi_dbi_stm32_fmc_data {
+};
+
+int mipi_dbi_stm32_fmc_command_write(const struct device *dev,
+				     const struct mipi_dbi_config *dbi_config, uint8_t cmd,
+				     const uint8_t *data_buf, size_t len)
+{
+	const struct mipi_dbi_stm32_fmc_config *config = dev->config;
+
+	uint32_t i = 0;
+
+	*config->register_addr = cmd;
+
+	while (i < len) {
+		uint16_t data = (uint16_t)data_buf[i];
+		*config->data_addr = data;
+		i += 1;
+	}
+
+	return 0;
+}
+
+static int mipi_dbi_stm32_fmc_write_display(const struct device *dev,
+					    const struct mipi_dbi_config *dbi_config,
+					    const uint8_t *framebuf,
+					    struct display_buffer_descriptor *desc,
+					    enum display_pixel_format pixfmt)
+{
+	const struct mipi_dbi_stm32_fmc_config *config = dev->config;
+	uint32_t i = 0;
+
+	while (i < desc->buf_size) {
+		uint16_t data = ((uint16_t)framebuf[i + 1U] << 8U) | (uint16_t)framebuf[i];
+		*config->data_addr = data;
+		i += 2;
+	}
+
+	return 0;
+}
+
+static int mipi_dbi_stm32_fmc_reset(const struct device *dev, uint32_t delay)
+{
+	const struct mipi_dbi_stm32_fmc_config *config = dev->config;
+	int ret;
+
+	if (config->reset.port) {
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_set_dt(&config->reset, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_msleep(delay);
+	return gpio_pin_set_dt(&config->reset, 0);
+}
+
+static int mipi_dbi_stm32_fmc_init(const struct device *dev)
+{
+	const struct mipi_dbi_stm32_fmc_config *config = dev->config;
+
+	if (config->reset.port) {
+		if (!gpio_is_ready_dt(&config->reset)) {
+			LOG_ERR("Reset GPIO device not ready");
+			return -ENODEV;
+		}
+
+		if (gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_INACTIVE)) {
+			LOG_ERR("Couldn't configure reset pin");
+			return -EIO;
+		}
+	}
+
+	if (config->power.port) {
+		if (!gpio_is_ready_dt(&config->power)) {
+			LOG_ERR("Power GPIO device not ready");
+			return -ENODEV;
+		}
+
+		if (gpio_pin_configure_dt(&config->power, GPIO_OUTPUT)) {
+			LOG_ERR("Couldn't configure power pin");
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static struct mipi_dbi_driver_api mipi_dbi_stm32_fmc_driver_api = {
+	.reset = mipi_dbi_stm32_fmc_reset,
+	.command_write = mipi_dbi_stm32_fmc_command_write,
+	.write_display = mipi_dbi_stm32_fmc_write_display,
+};
+
+#define MIPI_DBI_STM32_FMC_INIT(n)                                                                 \
+	static const struct mipi_dbi_stm32_fmc_config mipi_dbi_stm32_fmc_config_##n = {            \
+		.reset = GPIO_DT_SPEC_INST_GET_OR(n, reset_gpios, {}),                             \
+		.power = GPIO_DT_SPEC_INST_GET_OR(n, power_gpios, {}),                             \
+		.register_addr = (uint16_t *)FMC_BANK1_1,                                          \
+		.data_addr = (uint16_t *)(FMC_BANK1_1 + FMC_DATA_ADDR_OFFSET),                     \
```

Comment:
Maybe it is worth to add extra FMC API to get bank address based on bank `reg` and compute data address based on register address and RS signal I/O bit?

Reply:
I added a macro to compute the bank address. I recompute it from the base address of the bank1.
For the data address I don't really understand what you are asking ?

Reply:
> For the data address I don't really understand what you are asking ?

The data address is calculated as `[bank address] + (1 << [RS signal bit])` (_8-bit bus width_) and `[bank address] + (1 << ([RS signal bit] + 1))` (_16-bit bus width_).

For example, the `stm32l562e_dk` connects the panel over FMC via `BANK1` and LCD Register Select `A0` with 16-bit bus width. Therefore, `data_addr = FMC_BANK1_1 + (1 << (0 + 1))` where `0` is the RS signal I/O bit number. For `A1` it would be `1`st bit, the `A2` is `2`nd bit, and so on. That is why you have `FMC_DATA_ADDR_OFFSET=0x02` (`1 << (0+1) = 1 << 1 = 2`).

Unfortunately, there is no way to get such information from `pinctrl` (`fmc_a0_pf0`). Maybe the easiest way is to extend `st,mipi-dbi-fmc.yaml` with `lcd-register-select` enum.

@erwango, ideas?

Reply:
Okay, with your explanation and re-reading the datasheets I now understand how it all work.
Extending the binding seems to be the right solution ?

Reply:
>  Extending the binding seems to be the right solution ?

I am not against it. However, maybe someone has a better idea.

Reply:
Could we just use the `reg` property for this? It seems like this is indicating the address of the device, but please correct me if I'm missing something here

Reply:
@danieldegrasse is there an easy way to get child's reg without knowing child's label?

Reply:
I think the best option would probably be to pass the child reg in using the `slave` field of the `struct spi_config` present in the MIPI DBI API calls, since that is where a child device on a SPI bus would pass in its register address

Reply:
In Intel 8080 interface you have "Register Select (RS)" signal. The RS signal allows to differentiate instruction and data writes. That is called "LCD Register Select" in the FMC.

From the FMC perspective, the RS signal pin is configured in `pinctrl` and controlled depending on the address we write to. Therefore, we have to provide it in `pinctrl` and account for it in the data address.

It is not really a device address in itself.

Ideally, it must be taken from `pinctrl` to avoid providing such information more than once. But there is no easy way to do it.

Reply:
> Ideally, it must be taken from pinctrl to avoid providing such information more than once. But there is no easy way to do it.

Not doable indeed. A specific property is required.